packages/components/badge/src/Badge/Badge.styles.ts

```diff
@@ -83,9 +83,9 @@ export const getBadgeStyles = () => ({
   badgeText: css({
     color: 'currentcolor',
     lineHeight: 'inherit',
-    textTransform: 'lowercase',
+    textTransform: noTextTransforms ? 'none' : 'lowercase',
     '::first-letter': {
-      textTransform: 'uppercase',
+      textTransform: noTextTransforms ? 'none' : 'uppercase',
```

Comment:
Was the existing transforms trying to simulate `text-transform: capitalize` ? 

If yes then maybe it's better to replace two with one transform and change `noTextTransforms` to `textTransform: 'none' | 'capitalize` | ...`? Just something to think about.